stdlib/strscan.rb

```diff
@@ -297,6 +297,10 @@ def unscan
 
 private
   def anchor(pattern)
-    `new RegExp('^(?:' + pattern.toString().substr(1, pattern.toString().length - 2) + ')')`
+    %x{
+      var flags = pattern.toString().match(/\/([^\/]+)$/)
+      flags = flags && flags[1] || undefined
```

Comment:
@patrickkettner can you replace with the ternary operator `?:` which is slitghly more explicit?